### PR TITLE
pass token outside of SDK for server-to-server case

### PR DESCRIPTION
**Pull request by rmgogogo**
Pass token outside of SDK for advanced cases, e.x. server-to-server calls (KFP SDK in another server like CloudFunction).

We also can put all token-generation codes into SDK. However it would potentially still miss cases and for certain cases, better we allow pass-in token.

### CODE REVIEW

**Comment by rmgogogo** [comment]
fix #2920

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -125,7 +126,9 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
 
     # Obtain the tokens if it is IAP or inverse proxy.
     # client_id is only used for IAP, so when the value is provided, we assume it's IAP.
-    if client_id:
+    if other_token:
```
Comment (jessiezcc) [inline sdk/python/kfp/_client.py]: can u add some comments on what's the use case for other_token?

File: sdk/python/kfp/_client.py [hunk 2]
```diff
@@ -99,10 +99,11 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client
       other_client_id: The client ID used to obtain the auth codes and refresh tokens.
         Reference: https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_desktop_app.
       other_client_secret: The client secret used to obtain the auth codes and refresh tokens.
+      other_token: pass in token directly, it's used for cases better get token outside of SDK, e.x. GCP Cloud Functions.
```
Comment (Ark-kun) [inline sdk/python/kfp/_client.py]: Is `other_token` an official term?

**Comment by Bobgy** [comment]
So for cloud function, which token do we pass to the other_token? The service account token?

File: sdk/python/kfp/_client.py (hunk 2 above)
Comment (rmgogogo) [reply sdk/python/kfp/_client.py]: no, just followed the style of upper field

File: sdk/python/kfp/_client.py (hunk 1 above)
Comment (rmgogogo) [reply sdk/python/kfp/_client.py]: good point, I planned to put it to doc but keep quick sample here is better.

(don't know why PR routed to your queue, hope no interruption)

File: sdk/python/kfp/_client.py (hunk 2 above)
Comment (rmgogogo) [reply sdk/python/kfp/_client.py]: renamed to "existing_token" which I think you may like it.

File: sdk/python/kfp/_client.py [hunk 3]
```diff
@@ -123,10 +125,25 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
 
     token = None
 
-    # Obtain the tokens if it is IAP or inverse proxy.
-    # client_id is only used for IAP, so when the value is provided, we assume it's IAP.
-    if client_id:
-      token = get_auth_token(client_id, other_client_id, other_client_secret)
```
Comment (Ark-kun) [inline sdk/python/kfp/_client.py]: Why is this branch removed?

Comment (rmgogogo) [reply sdk/python/kfp/_client.py]: my mistake when merging codes, fixed. thanks

**Pull request merged**